pytorch_lightning/trainer/properties.py

```diff
@@ -139,6 +144,12 @@ def log_dir(self) -> Optional[str]:
         dirpath = self.accelerator.broadcast(dirpath)
         return dirpath
 
+    @property
+    def experiment_dir(self):
+        dirpath = self.logger.experiment_dir if self.logger is not None else self.default_root_dir
+        dirpath = self.accelerator.broadcast(dirpath)
```

Comment:
this broadcast should be done once so we're not calling this on every property access